### PR TITLE
Fix generated plugins.txt resource dependency

### PR DESCRIPTION
**Pull request by ChrisHegarty**
Running ESIntegTestCases in IntelliJ fails since the generated _plugins.txt_ resource file is not present.

One way to fix this is to add the generated-resources dir to the main sourceset. This will then be added to the class path when running tests in the IDE. But is requires that the generated-resources be present ( by, say, compiling gradle at least once beforehand).

### CODE REVIEW

**Comment by ChrisHegarty** [comment]
Example failure:

```
java.lang.NullPointerException
	at java.base/java.io.Reader.<init>(Reader.java:168)
	at java.base/java.io.InputStreamReader.<init>(InputStreamReader.java:112)
	at org.elasticsearch.common.io.Streams.readAllLines(Streams.java:185)
	at org.elasticsearch.common.io.Streams.readAllLines(Streams.java:180)
	at org.elasticsearch.plugins.PluginsService.getOfficialPlugins(PluginsService.java:212)
	at org.elasticsearch.plugins.PluginsService.getRuntimeInfos(PluginsService.java:194)
	at org.elasticsearch.plugins.PluginsService.<init>(PluginsService.java:150)
	at org.elasticsearch.plugins.MockPluginsService.<init>(MockPluginsService.java:38)
	at org.elasticsearch.node.MockNode.lambda$new$1(MockNode.java:104)
	at org.elasticsearch.node.Node.<init>(Node.java:388)
	at org.elasticsearch.node.MockNode.<init>(MockNode.java:104)
	at org.elasticsearch.node.MockNode.<init>(MockNode.java:87)
	at org.elasticsearch.test.InternalTestCluster.buildNode(InternalTestCluster.java:773)
	at org.elasticsearch.test.InternalTestCluster.reset(InternalTestCluster.java:1189)
	at org.elasticsearch.test.InternalTestCluster.beforeTest(InternalTestCluster.java:1086)
	at org.elasticsearch.test.ESIntegTestCase.lambda$beforeInternal$0(ESIntegTestCase.java:362)
```

**Review by rjernst — APPROVED** [review]
LGTM

**Review by breskeby — CHANGES_REQUESTED** [review]
We need to tweak the wiring and generation of the generated resources dir with the sourceSet. Added some hints to do so

**Comment by ChrisHegarty** [comment]
Thank you @breskeby - review comments adopted. Testing ...

File: server/build.gradle [hunk 1]
```diff
@@ -133,6 +132,7 @@ def generatePluginsList = tasks.register("generatePluginsList") {
 tasks.named("processResources").configure {
```
Comment (breskeby) [inline server/build.gradle:132]: this processResources configuration should be deleted. it's not required anymore IMO

Comment (breskeby) [reply server/build.gradle:132]: `sourceSets.main.output.dir(generatedResourcesDir)` needs to be brought back

**Review by breskeby — CHANGES_REQUESTED** [review]
suggested changes have not been applied

File: server/build.gradle [hunk 2]
```diff
@@ -120,10 +120,9 @@ def generatePluginsList = tasks.register("generatePluginsList") {
     Set<String> plugins = new TreeSet<>(project(':plugins').childProjects.keySet())
     plugins.remove('example')
 
-    File pluginsFile = new File(buildDir, 'generated-resources/plugins.txt')
+    File pluginsFile = new File(generatedResourcesDir, 'plugins.txt')
     processResources.from(pluginsFile)
     inputs.property('plugins', plugins)
-    outputs.file(pluginsFile)
```
Comment (breskeby) [inline server/build.gradle:126]: this should be kept imo

**Review by breskeby — APPROVED** [review]
lgtm

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

**Pull request merged**